core/src/main/scala/akka/kafka/internal/CommittingProducerSinkStage.scala

```diff
@@ -164,7 +168,18 @@ private final class CommittingProducerSinkStageLogic[K, V, IN <: Envelope[K, V,
     case CommittingProducerSinkStage.CommitNow => commit(Interval)
   }
 
-  private def collectOffset(count: Int, offset: Committable): Unit = {
+  private def collectOffset(count: Int, offset: Committable): Unit =
+    (stage.committerSettings.trigger, deferredOffset) match {
+      case (FirstObserved, _) =>
+        addToBatch(count, offset)
+      case (DeferToNextCommit, None) =>
```

Comment:
`isClosed(stage.in)` should trigger immediate committing.

Reply:
Got it.  I added that back for all cases.